app/ts/CreateMarketUI/components/CreateMarket.tsx

```diff
@@ -0,0 +1,200 @@
+import { useSignal } from '@preact/signals'
+import { approveErc20Token, createYesNoMarket } from '../../utils/utilities.js'
+import { OptionalSignal } from '../../utils/OptionalSignal.js'
+import { AccountAddress, EthereumAddress, EthereumQuantity, NonHexBigInt } from '../../types/types.js'
+import { AUGUR_CONTRACT, DAI_TOKEN_ADDRESS, GENESIS_UNIVERSE, REPV2_TOKEN_ADDRESS } from '../../utils/constants.js'
+import { addressString, decimalStringToBigint, isDecimalString } from '../../utils/ethereumUtils.js'
+
+interface CreateYesNoMarketProps {
+	maybeAccountAddress: OptionalSignal<AccountAddress>
+}
+
+const isValidDate = (dateStr: string): boolean => {
+	const regex = /^\d{4}-\d{2}-\d{2}$/
+	if (!regex.test(dateStr)) return false
+
+	const date = new Date(dateStr)
+	const [year, month, day] = dateStr.split('-').map(Number)
+
+	return (
+		date.getFullYear() === year &&
+		date.getMonth() + 1 === month &&
+		date.getDate() === day
+	)
+}
+
+const affiliateFeeOptions = [0, 1, 2, 3, 4, 5, 10, 15, 20, 25, 50, 75, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000].map((divisor) => ({
+	id: divisor,
+	name: divisor === 0 ? "0.00%" : `${ (100 / divisor).toFixed(2) }%`
+}))
+
+export const CreateYesNoMarket = ({ maybeAccountAddress }: CreateYesNoMarketProps) => {
+	const endTime = useSignal<string>('')
+	const marketCreatorFee = useSignal<string>('')
+	const affiliateValidator = useSignal<string>('0x0000000000000000000000000000000000000000')
```

Comment:
0x00 address fine for the default I think, just noting the functional one that exists is `0x2A256518DDdCe6E2e95B56a95991d4FA990Be659` that we could default to if the user actually wants to have an affiliate fee.